### PR TITLE
JDK-8286153: Remove redundant casts and other cleanup

### PR DESCRIPTION
**Pull request by jonathan-gibbons**
Please review some cleanup updates to address issues reported by an IDE.

The seeds for the change were a series of redundant casts, that have now all been removed.  Various other warnings and suggestions were made by the IDE for the affected files. There were a number of places with redundant type arguments, for which the general fix was in favor of using `var` instead of `<>`.

Some `switch` statements were converted to the enhanced `switch` form, which also revealed a couple of places where `RECORD` should have been added alongside `ENUM`.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 reviewer)

### Issue
 * [JDK-8286153](https://bugs.openjdk.java.net/browse/JDK-8286153): Remove redundant casts and other cleanup

### Reviewers
 * [Pavel Rappo](https://openjdk.java.net/census#prappo) (@pavelrappo - **Reviewer**) ⚠️ Review applies to [6ad78546](https://git.openjdk.java.net/jdk/pull/8543/files/6ad785461550707572be064d5d07a354b173f79a)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8543/head:pull/8543` \
`$ git checkout pull/8543`

Update a local copy of the PR: \
`$ git checkout pull/8543` \
`$ git pull https://git.openjdk.java.net/jdk pull/8543/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8543`

View PR using the GUI difftool: \
`$ git pr show -t 8543`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8543.diff">https://git.openjdk.java.net/jdk/pull/8543.diff</a>

</details>

### CODE REVIEW

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/MetaKeywords.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ public List<String> getMetaKeywords(TypeElement typeElement) {
             results.addAll(getMemberKeywords(utils.getFields(typeElement)));
             results.addAll(getMemberKeywords(utils.getMethods(typeElement)));
         }
-        ((ArrayList)results).trimToSize();
+        results.trimToSize();
```
Comment (pavelrappo) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/MetaKeywords.java:81]: I wonder if this method will look better this way:
```
    public List<String> getMetaKeywords(TypeElement typeElement) {
        var results = new ArrayList<String>();
        ...
        results.trimToSize();
        return results;
    }
```

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/MetaKeywords.java [hunk 2]
```diff
@@ -122,7 +122,7 @@ public List<String> getMetaKeywordsForModule(ModuleElement mdle) {
      * Get the overview keywords.
      */
     public List<String> getOverviewMetaKeywords(String title, String docTitle) {
-         List<String> result = new ArrayList<>(1);
+        List<String> result = new ArrayList<>(1);
```
Comment (pavelrappo) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/MetaKeywords.java:125]: Consider using `var` similarly to my suggestion above. Alternatively, we can use two calls to `List.of(e)` and one call to `List.of()`.

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/MetaKeywords.java [hunk 3]
```diff
@@ -153,7 +153,7 @@ protected List<String> getMemberKeywords(List<? extends Element> members) {
                 results.add(membername);
             }
         }
-        ((ArrayList)results).trimToSize();
+        results.trimToSize();
```
Comment (pavelrappo) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/MetaKeywords.java:156]: Same suggestion for `var` as above.

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/ElementsTable.java [hunk 4]
```diff
@@ -272,7 +270,7 @@ public Set<? extends Element> getSpecifiedElements() {
      * A module is fully included,
      *   - is specified on the command line --module
      *   - is derived from the module graph, that is, by expanding the
-     *     requires directive, based on --expand-requires
+     *     'requires' directive, based on --expand-requires
```
Comment (pavelrappo) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/ElementsTable.java:273]: Thanks for using these clarifying quotes!

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/ElementsTable.java [hunk 5]
```diff
@@ -1061,14 +1059,11 @@ public Boolean visitType(TypeElement e, Void p) {
...
-                    switch(enclosing.getKind()) {
-                        case PACKAGE:
-                            return specifiedPackageElements.contains((PackageElement)enclosing);
-                        case CLASS: case INTERFACE: case ENUM: case ANNOTATION_TYPE:
-                            return visit((TypeElement) enclosing);
-                        default:
-                            throw new AssertionError("unknown element: " + enclosing);
-                    }
+                    return switch (enclosing.getKind()) {
+                        case CLASS, ENUM, RECORD, INTERFACE, ANNOTATION_TYPE -> visit(enclosing);
```
Comment (pavelrappo) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/ElementsTable.java:1063]: Whoa! `RECORD` was missing. Does it make sense to accompany this PR with a small test that crashes javadoc with a type nested in a non-included record?

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/ElementsTable.java [hunk 6]
```diff
@@ -1208,40 +1203,25 @@ static class ModifierFilter {
...
-                        accessValue  = options.showPackagesAccess();
-                        break;
-                    case MODULE:
-                        accessValue  = options.showModuleContents();
-                        break;
-                    default:
-                        throw new AssertionError("unknown element: " + kind);
-
-                }
+                accessValue = switch (kind) {
```
Comment (pavelrappo) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/ElementsTable.java:1206]: It feels awkward when adjacent `switch` statements use different formatting.

Comment (pavelrappo) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/ElementsTable.java:1206]: Correction: `switch` _expressions_ in this case.

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/MetaKeywords.java (hunk 1 above)
Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/MetaKeywords.java:81]: Agreed; I was just trying to limit the cleanup. `var` is good.

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/MetaKeywords.java (hunk 3 above)
Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/MetaKeywords.java:156]: 👍

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/ElementsTable.java (hunk 6 above)
Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/ElementsTable.java:1206]: will examine

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/MetaKeywords.java (hunk 2 above)
Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/MetaKeywords.java:125]: Most of these cases seem better with appropriate variants of `List.of`

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/ElementsTable.java (hunk 5 above)
Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/ElementsTable.java:1063]: I've reviewed all uses of `ENUM`. All other places in javadoc look OK for having a sibling case for `RECORD`.

Filed [JDK-8286205](https://bugs.openjdk.java.net/browse/JDK-8286205) for more tests.

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/ElementsTable.java (hunk 6 above)
Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/ElementsTable.java:1206]: fixed

**Review by pavelrappo — APPROVED** [review]

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/ClassUseMapper.java [hunk 7]
```diff
@@ -221,7 +221,7 @@ public ClassUseMapper(BaseConfiguration configuration, ClassTree classtree) {
             for (VariableElement fd : fields) {
                 mapTypeParameters(classToFieldTypeParam, fd, fd);
                 mapAnnotations(annotationToField, fd, fd);
-                SimpleTypeVisitor9<Void, VariableElement> stv = new SimpleTypeVisitor9<Void, VariableElement>() {
+                 var stv = new SimpleTypeVisitor9<Void, VariableElement>() {
```
Comment (liach) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/ClassUseMapper.java:224]: Extra space before `var`

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java [hunk 8]
```diff
@@ -1528,8 +1525,9 @@ private DocCollator(Locale locale, int strength) {
             instance = createCollator(locale);
             instance.setStrength(strength);
 
-            keys = new LinkedHashMap<String, CollationKey>(MAX_SIZE + 1, 0.75f, true) {
+            keys = new LinkedHashMap<>(MAX_SIZE + 1, 0.75f, true) {
```
Comment (liach) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java:1528]: This map's initial capacity requires it to resize before it starts evicting, see https://github.com/openjdk/jdk/blob/7ebc4bce930e7d406f1e68c8bd09ac1dc3a21e5a/src/java.base/share/classes/java/util/HashMap.java#L2563

Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java:1528]: Not in this PR.

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/ClassUseMapper.java (hunk 7 above)
Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/ClassUseMapper.java:224]: Thanks, will fix.

**Pull request closed without merging**